### PR TITLE
Skip failed LFS objects and continue when migrating LFS

### LINKED ISSUE #31137: Migrating repo with LFS causes 500 error
### Description

https://gitea.com/Shuenhoy/vtk-m is migrated from https://gitlab.kitware.com/vtk/vtk-m. 
Accessing the `data` folder will trigger 500 error.

Whether selecting `migrate LFS files` is the same.
In the demo site, https://demo.gitea.com/Shuenhoy/vtk-m will not trigger 500 error, but it doesn't treat the files as LFS files at all. 

### Gitea Version

1.22.0

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

image

(On my instance, there is an extra message)
```
An error occurred:

getFileReader, LFS Meta object does not exist
```

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Docker

### Database

None

### PR DESCRIPTION
**Pull request by 6543**
Partially fix https://github.com/go-gitea/gitea/issues/31137.

Continue #31623 ...

When migrating LFS objects, if there's any object that failed (like some objects are losted, which is not really critical), Gitea will stop migrating LFS immediately but treat the migration as successful.

This PR makes it ignore failed objects and continue migrating. It may not be the best way to handle it, but a better way.

And to ensure data integrity of LFS objects, we may need to add an option to fail the migration if there's any error in the feature.

### CODE REVIEW

**Comment by 6543** [comment]
PS: the whole code is quite complex and not good to maintain ... it should be refactored into smaler chunks and probably into it's own submodule(s)

File: modules/repository/repo.go [hunk 1]
```diff
@@ -233,28 +244,32 @@ func StoreMissingLfsObjectsInRepository(ctx context.Context, repo *repo_model.Re
...
-			_, err := git_model.NewLFSMetaObject(ctx, repo.ID, pointerBlob.Pointer)
+			m, err := git_model.NewLFSMetaObject(ctx, repo.ID, pointerBlob.Pointer)
 			if err != nil {
 				log.Error("Repo[%-v]: Error creating LFS meta object %-v: %v", repo, pointerBlob.Pointer, err)
 				return err
 			}
+			if m.Existing {
+				log.Trace("Repo[%-v]: LFS meta object %-v was already present", repo, m)
+			}
+			continue
```
Comment (6543) [inline modules/repository/repo.go:255]: this continue is not needed but make the code more redable ... at least a bit

**Comment by wolfogre** [comment]
Hmm, I would say it is a different idea from what I had in mind.

`ErrLFSDownload` is not clear enough to indicate the error.

I think I have to write my idea down to discuss ...

**Comment by lunny** [comment]
replaced by #31702

**Pull request closed without merging**